main.cpp

```diff
@@ -261,247 +367,365 @@ struct CoatRack
         void provideStyle(bool matchesJacket);
     };
 
-    //1) mount to wall
-    float mountRack(float wallHeight, bool addedSupport);
-    //2) hang hats
+    float mountRack(double wallHeight, bool addedSupport);
     void hangHats(Hat myHat1, int hookToHang, int hatsPerHook);
-    //3) hang coats
     void hangCoats(int coatsToHang, int coatsPerHook);
 
     Hat myHat1;
     Hat myHat2;
 };
-/*
-Thing 4) Remote Control
-5 properties:
-    1) on/off switch
-    2) contains batteries
-    3) left/right buttons
-    4) up/down buttons
-    5) weight of object
-3 things it can do:
-    1) change channel
-    2) switch mode
-    3) turn tv off
- */
+
+void CoatRack::Hat::sitOnHead(float howLongToWear, bool doesFit)
+{
+    if (diameter > 2.4f)
+    {
+        doesFit = true;
+    }
+    howLongToWear = 32.5f;
+}
+
+void CoatRack::Hat::containsRabbit(bool isRabbitInHat)
+{
+    if (isRabbitInHat == true)
+    {
+        sitOnHead(0.0f, false);
+    }
+}
+
+void CoatRack::Hat::provideStyle(bool matchesJacket)
+{
+    if (color == "blue")
+    {
+        matchesJacket = true;
+    }
+}
+
+float CoatRack::mountRack(double wallHeight, bool addedSupport)
+{
+    if (addedSupport == true)
+    {
+        wallHeight *= rackWidth;
+    }
+    return 0;
+}
+
+void CoatRack::hangHats(Hat thisHat, int hookToHang, int hatsPerHook)
+{
+    thisHat.containsRabbit(false);
+    hatsPerHook += hookToHang;
+}
+
+void CoatRack::hangCoats(int coatsToHang, int coatsPerHook)
+{
+    if (coatsToHang > 6)
+    {
+        coatsPerHook = 2;
+    }
+}
+
+//==========================================================
+
 struct RemoteControl
 {
-    // 1) on/off switch
     int onOffSwitch = 1;
-    // 2) contains batteries
     bool containsBatteries = true;
-    // 3) left/right buttons
     bool leftRightButtons = false;
-    // 4) up/down buttons
     bool upDownButtons = true;
-    // 5) weight of object
     float objectWeight = 0.14f;
 
-    // 1) change channel
     int changeChannel(int channel);
-    // 2) switch mode
     void switchMode(bool hdmi);
-    // 3) turn tv off
     int turnOff(bool isButtonDown);
 };
-/*
-Thing 5) Channel Strip
-5 properties:
-    1) number of knobs
-    2) number of faders
-    3) max range of gain knob
-    4) text size used on control labels
-    5) number of inputs
-3 things it can do:
-    1) apply adjustments to channel
-    2) accept input of microphone
-    3) accept input of audio cable
- */
+
+int RemoteControl::changeChannel(int channel)
+{
+    if (leftRightButtons == true)
+    {
+        channel += 1;
+    }
+    return channel;
+}
+
+void RemoteControl::switchMode(bool hdmi)
+{
+    if (changeChannel(4))
+    {
+        hdmi = true;
+    }
+    
+}
+
+int RemoteControl::turnOff(bool isButtonDown)
+{
+    if (onOffSwitch == 1 && isButtonDown == true)
+    {
+        changeChannel(false);
```

Comment:
don't use `true` or `false` to indicate `1` or `0`.  be explicit.
At the same time, don't use `1` or `0` to indicate `true` or `false`.  Make your code as easy to read as possible.
- [ ] revise